File: src/main/java/dev/slne/discord/listener/message/AbstractMessageListener.java
package dev.slne.discord.listener.message;

import dev.slne.discord.spring.service.ticket.TicketService;
import dev.slne.discord.ticket.Ticket;
import java.util.Optional;
import lombok.AllArgsConstructor;
import net.dv8tion.jda.api.entities.channel.Channel;
import net.dv8tion.jda.api.entities.channel.ChannelType;
import net.dv8tion.jda.api.events.message.GenericMessageEvent;
import net.dv8tion.jda.api.hooks.ListenerAdapter;
import org.jetbrains.annotations.NotNull;

@AllArgsConstructor
public abstract class AbstractMessageListener<T extends GenericMessageEvent> extends
    ListenerAdapter {

  private final TicketService ticketService;

  protected void handleEvent(T event, Ticket ticket) {
  }

  protected final void processEvent(@NotNull T event) {
    getTicketByChannel(event.getChannel()).ifPresent(ticket -> handleEvent(event, ticket));
  }

  protected Optional<Ticket> getTicketByChannel(Channel channel) {
    if (!channel.getType().equals(ChannelType.TEXT)) {
      return Optional.empty();
    }

    return ticketService.getTicketByChannelId(channel.getId());
  }
}

File: src/main/kotlin/dev/slne/discord/listener/message/AbstractMessageListener.kt
package dev.slne.discord.listener.message

import dev.slne.discord.spring.service.ticket.TicketService
import dev.slne.discord.ticket.Ticket
import net.dv8tion.jda.api.entities.channel.Channel
import net.dv8tion.jda.api.entities.channel.ChannelType

abstract class AbstractMessageListener {

    protected fun getTicketByChannel(channel: Channel): Ticket? {
        if (channel.type != ChannelType.GUILD_PRIVATE_THREAD) {
            return null
        }

        return TicketService.getTicketByChannelId(channel.id)
    }
}
